webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/dialog/OpenDocumentDialogPanel.java

```diff
@@ -73,189 +68,215 @@
-            }).add(new AjaxEventBehavior("dblclick")
+                
+                docListChoice.setChoices(listDocuments());
+                docListChoice.setDefaultModel(Model.of());
+                aTarget.add(buttonsForm);
+                aTarget.add(docListChoice);
+            }
+        });
+        
+        if (aBModel.isProjectLocked()) {
+            projectListChoice.setVisible(false);
+        }
+        
+        return projectListChoice;
+    }
+    
+    private OverviewListChoice<DecoratedObject<User>> createUserListChoice()
+    {
+        DecoratedObject<User> currentUser = DecoratedObject.of(userRepository.getCurrentUser());
+        currentUser.setColor(CURRENT_USER_COLOR);
+        userListChoice = new OverviewListChoice<>("user", Model.of(currentUser), listUsers());
+        userListChoice.setChoiceRenderer(new ChoiceRenderer<DecoratedObject<User>>()
+        {
+
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public Object getDisplayValue(DecoratedObject<User> aUser)
             {
+                User user = aUser.get();
+                String username = defaultIfEmpty(aUser.getLabel(), user.getUsername());
+                return username + (user.isEnabled() ? "" : " (login disabled)");
+            }
+        });
+        userListChoice.setOutputMarkupId(true);
+        userListChoice.add(new OnChangeAjaxBehavior()
+        {
+            private static final long serialVersionUID = 1L;
 
-                private static final long serialVersionUID = 1L;
+            @Override
+            protected void onUpdate(AjaxRequestTarget aTarget)
+            {
+                docListChoice.setChoices(listDocuments());
+                docListChoice.setDefaultModel(Model.of());
+                aTarget.add(buttonsForm);
+                aTarget.add(docListChoice);
+            }
+        }).add(visibleWhen(() -> state.getMode().equals(Mode.ANNOTATION)
+                && isManagerForListedProjects()));
 
-                @Override
-                protected void onEvent(final AjaxRequestTarget aTarget)
-                {
-                    selectedProject = getModelObject().projectSelection;
-                    // Remove selected document from other project
-                    selectedDocument = null;
-                    aTarget.add(documentSelection.setOutputMarkupId(true));
-                }
-            });
-        }
+        return userListChoice;
     }
 
-    private class SelectionModel
-        implements Serializable
+    /**
+     * Check if current user is manager for any of the listed projects
+     */
+    private boolean isManagerForListedProjects()
     {
-        private static final long serialVersionUID = -1L;
-
-        private Project projectSelection;
-        private SourceDocument documentSelection;
+        return projectService.isManager(projectListChoice.getModelObject().get(),
+                userRepository.getCurrentUser())
```

Comment:
The selected project can also be the one saved in the annotatorstate given as a constructor parameter. Theoretically, this project might not be included in the given projects. So, I added this check here to make sure.

Reply:
Did you observe this happening in practice or is it a theoretical construction? I'd say that under normal circumstances the list of accessible projects *must* include any project in which the current user is a manager, otherwise there would be a bug in the listing of accessible projects.

Reply:
This is theoretical, I did not observe it.